### PR TITLE
Address deprecation warnings

### PR DESCRIPTION
**Pull request by Palleas**
- Rename `Lock` to `NIOLock`
- Replace uses of `NIOAtomic` with `ManagedAtomic`
- Bump copyright

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -25,11 +25,14 @@ let package = Package(
...
         .package(url: "https://github.com/apple/swift-nio.git", from: "2.0.0"),
         .package(url: "https://github.com/apple/swift-docc-plugin", from: "1.0.0"),
+        .package(url: "https://github.com/apple/swift-atomics.git", from: "1.0.0"),
     ],
     targets: [
         .target(name: "StatsdClient", dependencies: [
             .product(name: "CoreMetrics", package: "swift-metrics"),
             .product(name: "NIO", package: "swift-nio"),
+            .product(name: "Atomics", package: "swift-atomics"),
+
```
Comment (tomerd) [inline Package.swift]: nit: extra line

**Comment by tomerd** [comment]
swift-atomics is not available on older versions (< 5.5) so we may need to live with the deprecation warnings or #if relevant code. I think we addressed something similar in https://github.com/swift-server/swift-service-lifecycle

see https://github.com/swift-server/swift-service-lifecycle/blob/main/Sources/Lifecycle/Atomics.swift

File: scripts/soundness.sh [hunk 2]
```diff
@@ -19,7 +19,11 @@ here="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
 function replace_acceptable_years() {
     # this needs to replace all acceptable forms with 'YEARS'
+<<<<<<< HEAD
+    sed -e 's/20[12][901]-202[0123]/YEARS/' -e 's/2019/YEARS/' -e 's/202[0123]/YEARS/'
+=======
     sed -e 's/20[12][901]-202[0123]/YEARS/' -e 's/2019/YEARS/' -e 's/202[012]/YEARS/'
+>>>>>>> origin/main
```
Comment (tomerd) [inline scripts/soundness.sh]: accidental commit?

Comment (Palleas) [reply scripts/soundness.sh]: Yep!

**Review by tomerd — APPROVED** [review]

**Pull request merged**